python/pyarrow/table.pxi

```diff
@@ -4859,6 +4859,91 @@ cdef class Table(_Tabular):
             output_type=Table
         )
 
+    def join_asof(self, right_table, on, by, tolerance, right_on=None, right_by=None):
+        """
+        Perform an asof join between this table and another one.
+
```

Comment:
I would still like to see a bit more expanded explanation (apart from the individual keyword explanations) about _what_ and asof join exactly is.

Something indicating it does 1) an inexact join, 2) on a sorted dataset, 3) potentially first joining on other attributes, and 4) typically useful for time series data that are not perfectly aligned. Are that the most relevant characteristics?

Reply:
https://github.com/apache/arrow/pull/34234/commits/e6da4df5cae7464439ae51c92ca1f43012a331a1